spec/erb_lint/linters/hard_coded_string_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe ERBLint::Linters::HardCodedString do
+  let(:linter_config) do
+    described_class.config_schema.new
+  end
+  let(:file_loader) { ERBLint::FileLoader.new('.') }
+  let(:linter) { described_class.new(file_loader, linter_config) }
+  let(:processed_source) { ERBLint::ProcessedSource.new('file.rb', file) }
+  subject(:offenses) { linter.offenses(processed_source) }
+
+  context 'when file contains hard coded string' do
+    let(:file) { <<~FILE }
+      <span> Hello </span>
+    FILE
+
+    it { expect(subject).to eq [arbitrary_error_message(6..12, 'Hello ')] }
+  end
+
+  context 'when file contains nested hard coded string' do
+    let(:file) { <<~FILE }
+      <span class="example">
+        <div id="hero">
+          <span id="cat"> Example </span>
+        </div>
+      </span>
+    FILE
+
+    it { expect(subject).to eq [arbitrary_error_message(60..68, 'Example ')] }
+  end
+
+  context 'when file contains a mix of hard coded string and erb' do
+    let(:file) { <<~FILE }
+      <span><%= foo %> Example </span>
+    FILE
+
+    it { expect(subject).to eq [arbitrary_error_message(6..24, '<%= foo %> Example ')] }
+  end
+
+  context 'when file contains hard coded string nested inside erb' do
+    let(:file) { <<~FILE }
+      <span>
+        <% foo do %>
+          <span> Example </span>
+        <% end %>
+      </span>
+    FILE
+
+    it { expect(subject).to eq [arbitrary_error_message(32..40, 'Example ')] }
+  end
+
+  context 'when file contains multiple hard coded string' do
+    let(:file) { <<~FILE }
+      <span> Example </span>
+      <span> Foo </span>
+      <span> Test </span>
+    FILE
+
+    it 'find all offenses' do
+      expect(subject).to eq [
+        arbitrary_error_message(6..14, 'Example '),
+        arbitrary_error_message(29..33, 'Foo '),
+        arbitrary_error_message(48..53, 'Test ')
+      ]
+    end
+  end
+
+  context 'when file does not contain any hard coded string' do
+    let(:file) { <<~FILE }
+      <span class="example">
+        <div id="hero">
+          <span id="cat"> <%= t(:hello) %> </span>
+        </div>
+      </span>
+    FILE
+
+    it { expect(subject).to eq [] }
+  end
+
+  context 'when file contains irrelevant hard coded string' do
+    let(:file) { <<~FILE }
+      <span class="example">
+        %
```

Comment:
I'm not sure I understand this use case. I think you might miss stray characters like forgotten `.` at the end of a sensence which should be caught?

Reply:
A false positive can happen if someone does this.
```erb
"<%= discounted %>%" 
# Would be rendered i.e. as 18%
```

Reply:
It might makes more sense to provide guidance towards creating helpers or using existing ones. For example https://apidock.com/rails/ActionView/Helpers/NumberHelper/number_to_percentage is the right one for `%` use case. That would allow us to really leave no text in html files other than whitespace. WDYT?